src/plugins/PluginSystem.cpp

```diff
@@ -119,6 +119,35 @@ void CPluginSystem::unloadAllPlugins() {
         unloadPlugin(p.get(), false); // Unload remaining plugins gracefully
 }
 
+std::vector<std::string*> CPluginSystem::updateConfigPlugins(const std::vector<std::string>& plugins, bool& changed) {
+    std::vector<std::string*> failures;
+
+    // unload all plugins that are no longer present
+    for (auto& p : m_vLoadedPlugins | std::views::reverse) {
+        if (p->m_bLoadedWithConfig && std::find(plugins.begin(), plugins.end(), p->path) == plugins.end()) {
+            Debug::log(LOG, "Unloading plugin %s which is no longer present in config", p->path.c_str());
+            unloadPlugin(p.get(), false);
```

Comment:
reallocates m_vLoadedPlugins, segfault.

Reply:
Where is the realloc? If its the `std::erase_if` at the end of `CPluginSystem::unloadPlugin` could you explain how the call in `CPluginSystem::unloadAllPlugins` is not ub? (I copied from there)